---
title: "Introduction to PXDataset2 object"
author: "Mohieddin Jafari"
date: "`r format(Sys.time(), '%Y-%m-%d %H:%M:%S')`"
output: html_document
---
## Resources
Bioconductor packages tagged by "proteomics":
https://bioconductor.org/packages/release/BiocViews.html#___Proteomics

```{r setup, include=TRUE, echo=TRUE, eval=FALSE}
knitr::opts_chunk$set(echo = FALSE)
## install.package("BiocManager")

#library(BiocManager)

# Bioconductor packages
# if (!requireNamespace("BiocManager", quietly = TRUE))
#     install.packages("BiocManager")
# 
# BiocManager::install("tidyverse")
# BiocManager::install("factoextra")
# BiocManager::install("msdata")
# BiocManager::install("mzR")
# BiocManager::install("rhdf5")
# BiocManager::install("rpx")
# BiocManager::install("MsCoreUtils")
# BiocManager::install("QFeatures")
# BiocManager::install("Spectra")
# BiocManager::install("ProtGenerics")
# BiocManager::install("PSMatch")
# BiocManager::install("pheatmap")
# BiocManager::install("limma")
# BiocManager::install("MSnID")
# BiocManager::install("RforMassSpectrometry/SpectraVis")
# 
# install.packages("tidyverse")
```

## Review some points about MS data
- The continuous workflow of separating-separating-separating!

- Three dimensions: Retention time of analyser elution - M/Z - Measurement of Intensity

**Spectrum is the unit of mass spectrometry data**


*TIC (total ion chromatogram) plot*: The intensities of all ions in each mass spectrometer scan (within a given time period) are summed and plotted as a function of time.
*Base peak chromatogram plot*: The intensity of the base peak (the most intense ion in each mass spectrum) is plotted as a function of time. (Note: do not confuse the base peak with the baseline, which represents the constant signal from the background level of the instrument.)
*XIC or EIC plot*: The intensity of a specific analyte at a chosen m/z value (either total or base peak intensity) is isolated and plotted as a function of time.
![](TIC_XIC.png)

## Getting data
Get to know the ProteomeXchange Consortium: Established to create globally coordinated, standardized pipelines for data submission and dissemination across major proteomics repositories, the Consortium promotes open data policies in proteomics.

`proteomexchange.org`

```{r, message=FALSE}
#Direct accessing data - this chunk is for checking the file types in the datasets and downloading only the mzML file that we need for this analysis
library("rpx")  #this package accesses the PRIDE dataset directly
px <- PXDataset("PXD022816")  #has 11 files in it
pxget(px, grep("mzID", pxfiles(px))[1:3])    #This line uses the rpx package to download files from a ProteomeXchange (PX) dataset, specifically the first 3 files that contain "mzID" in their names.
pxget(px, grep("mzML", pxfiles(px))[1:3])

px1 <- PXDataset("PXD000001")  # this is a smaller dataset was two files 
#px contains some number of files, Use 'pxfiles(.)' to see all files in the px.
pxfiles(px, as.vector = TRUE) |> print()
px #going forward we use the px of PXDataset 022816, this is the actual dataset for this experiment, the PXD00001 was just a test of a smaller dataset
fn <- "TMT_Erwinia_1uLSike_Top10HCD_isol2_45stepped_60min_01-20141210.mzML"   #we will eventually extract this file which is one of the 11 files from the "px" the 1st one. In my case, I have downloaded the mzML file that I will use. But to practice with this small dataset 000001, I will assume this mgf.gz file below is the mzML file
fn1 <- "PRIDE_Exp_Complete_Ac_22134.pride.mgf.gz"
mzf <- pxget(px, fn)
mzf <- pxget(px1, fn1)

pxtax(px1) # taxonomic name
pxref(px1) # bibliography ref

pxfiles(px1) # file names of the object
pxfiles(px1, as.vector = TRUE) |> print()

pxget(px, "erwinia_carotovora.fasta")  #This is supposed to extract the file number 11 from px
#in my case, since I can only access px1, I will extract one of the two files 
pxget(px1, "PRIDE_Exp_Complete_Ac_22134.pride.mztab.gz")
pxget(px, "README.txt") # a vector defining the files to be downloaded
# Some files are accessible remotly and some of them from local cashing mechanism.
(f <- pxget(px, "TMT_Erwinia_1uLSike_Top10HCD_isol2_45stepped_60min_01-20141210.mzML"))
basename(f)
```

## MS example data: msdata
```{r}
# Various Mass Spectrometry raw data example files for proteomics and metabolomics
library(msdata) 
proteomics()
ident()
quant()
```
### Quiz 1 (GA_4)
Visit `gosocrative.com` and enter room name `FIN2025`

## Raw MS data
- We need to convert binary-vendor-specific formats to open formats such as mzML, mzXML.
- mzML is the most recent format.
- proteowizard msconvert: https://proteowizard.sourceforge.io/ or search in Google msconvert GUI
- ThermoRawFileParser: https://github.com/compomics/ThermoRawFileParser

- To check all MS-related data format:
https://en.wikipedia.org/wiki/Mass_spectrometry_data_format

## MS data from R prospective
https://rformassspectrometry.github.io/book/img/F02-3D-MS1-MS2-scans-100-1200-lattice.png
---->>>>
https://rformassspectrometry.github.io/book/img/raw.png


gtg
